src/time_steppers.jl

```diff
@@ -410,9 +410,9 @@ Compute the vertical velocity w by integrating the continuity equation downwards
 function compute_w_from_continuity!(grid::AbstractGrid, U)
     @loop for j in (1:grid.Ny; (blockIdx().y - 1) * blockDim().y + threadIdx().y)
         @loop for i in (1:grid.Nx; (blockIdx().x - 1) * blockDim().x + threadIdx().x)
-            @inbounds U.w[i, j, 1] = 0
-            @unroll for k in 2:grid.Nz
-                @inbounds U.w[i, j, k] = U.w[i, j, k-1] + grid.Δz * ∇h_u(i, j, k-1, grid, U.u, U.v)
+            @inbounds U.w[i, j, grid.Nz] = 0
```

Comment:
This is incorrect --- remember that `Nz` is inside the domain. 

I think it'd be clearer to integrate from the bottom upwards; eg start at `k=1` and integrate up to `Nz`.

Otherwise, you should set the value of `w` at `Nz+1` (the top surface), and integrate downwards.

Reply:
To integrate upwards, write

```julia
U.w[i, j, 1] = 0
@unroll for k in 2:grid.Nz
    @inbounds U.w[i, j, k] = U.w[i, j, k-1] - grid.Δz * ∇h_u(i, j, k, grid, U.u, U.v)
end
```

Reply:
In summary, I think this function will work either way as long as you ensure that

1. You integrate over *all* interior grid points --- either `2:grid.Nz`, or `grid.Nz:-1:2` --- and

2. You refrain from setting the value of `w` at either top or bottom, since ensuring the correct value of `w` along the boundary is the job of `fill_halo_regions!` 

(Point 2 is not strictly necessary at this point since we only support `NoPenetration` boundary conditions, but will mean that adding new `Periodic` / Flow` / `Penetration` z-boundary conditions for vertical velocity will work out of the box without having to change this function.)

Reply:
I think we should be able to explain why integrating from the bottom up does not work.

Reply:
I agree. Perhaps I opened this PR too early as I'm still debugging. Will revisit once I've worked on other parts of this PR.

Reply:
@jm-c suggested that integrating from the bottom upwards is preferable as it will also work in the case that we decide to implement a free surface.